packages/engine/src/services/article-image-service.ts

```
@@ -0,0 +1,181 @@
+import { fal } from '@fal-ai/client';
+import { articleCover, renderPrompt } from '../prompts';
+
+interface FalImage {
+  url: string;
+  width?: number;
+  height?: number;
+  content_type?: string;
+}
+
+interface FalResponse {
+  data: {
+    images: FalImage[];
+  };
+}
+
+interface ArticleImageParams {
+  title: string;
+  summary: string;
+  category?: string;
+}
+
+/**
+ * Initialize the fal.ai client with API key
+ * Should be called once at startup
+ */
+export function initFalClient(): boolean {
+  const falKey = process.env.FAL_KEY;
+  if (!falKey) {
+    logger.warn(
+      'FAL_KEY not found - article image generation disabled',
+      {},
+      'ArticleImageService'
+    );
+    return false;
+  }
+
+  fal.config({
+    credentials: falKey,
+  });
+
+  return true;
+}
+
+/**
+ * Check if image generation is available
+ */
+export function isImageGenerationAvailable(): boolean {
+  return Boolean(process.env.FAL_KEY);
+}
+
+/**
+ * Generate a cover image for an article (best-effort, non-blocking)
+ *
+ * This function never throws - errors are logged and null is returned.
+ * This ensures image generation failures don't block tick execution.
+ *
+ * @param params - Article details for image generation
+ * @returns URL of the generated image, or null if generation fails
+ */
+export async function generateArticleImage(
+  params: ArticleImageParams
+): Promise<string | null> {
+  if (!isImageGenerationAvailable()) {
+    logger.debug(
+      'Skipping article image generation - FAL_KEY not available',
+      { title: params.title },
+      'ArticleImageService'
+    );
+    return null;
+  }
+
+  const prompt = renderPrompt(articleCover, {
+    title: params.title,
+    summary: params.summary,
+    category: params.category || 'general',
+  });
+
+  logger.debug(
+    'Generating article cover image',
+    { title: params.title, category: params.category },
+    'ArticleImageService'
+  );
+
+  let result: FalResponse;
+  try {
```

Note: this introduces a try/catch. Repo guidelines (`.cursorrules`) say to avoid try/catch / fail fast. If that rule is still strict, consider documenting this as an exception or moving the error handling up a layer.